### ISSUE TITLE
Error when trying to load module using systemjs

### ISSUE THREAD

**Issue by cck0001** [AndyMeps/ng2-dropdown-multiselect#10]
I am using systemjs to load modules with angular2 and have added the following for the ng2-dropdown-multiselect package to it's configuration:

map: {
...
        'ng2-dropdown-multiselect':   'npm:ng2-dropdown-multiselect/lib',
...
}

packages: {
...
        'ng2-dropdown-multiselect':   { format: 'cjs', main: 'index.js', defaultExtension: 'js' },
...
}

At runtime I get the following error trying to load the module:

zone.js:1382 GET http://localhost:3000/node_modules/ng2-dropdown-multiselect/lib/src/components.js 404 (Not Found)
and 
zone.js:1382 GET http://localhost:3000/node_modules/ng2-dropdown-multiselect/lib/src/models.js 404 (Not Found)

Perhaps I am doing something wrong and I noticed you are not using systemjs in your demo. Can you provide documentation on how to get this to work with systemjs.

**Comment by AndyMeps** [comment]
Hi cck0001,

Try something like this:

```javascript
var map = {
    …
    'ng2-dropdown-multiselect': 'npm:ng2-dropdown-multiselect'
};

var packages = {
    …
    'ng2-dropdown-multiselect': { main: 'lib/index.js', defaultExtension: 'js', map: {
            './lib/src/components': './lib/src/components/index.js',
            './lib/src/models': './lib/src/models/index.js'
     }}
};
```

This is a working example from my ng2-color-picker app, it may need tweaking slightly. I'll improve integration with systemjs in the future.

Regards,

Andy

**Comment by cck0001** [comment]
That put me on the right track and a slight variant worked:'ng2-dropdown-multiselect':   { main: './lib/index.js', defaultExtension: 'js',
                                map: {
                                    './lib/src/components.js': './lib/src/components/index.js',
                                    './lib/src/models.js': './lib/src/models/index.js'
                                }
                              },The initial code gave me an error about using a recursive map is not allowed. I believe that was because ./lib/src/components/index.js starts with ./lib/src/components

Thanks for your help and I look forward to using your very nice component.
Chris      From: Andy Mepham <notifications@github.com>
 To: AndyMeps/ng2-dropdown-multiselect <ng2-dropdown-multiselect@noreply.github.com> 
Cc: cck0001 <cck0001@yahoo.com>; Author <author@noreply.github.com>
 Sent: Friday, January 13, 2017 11:58 AM
 Subject: Re: [AndyMeps/ng2-dropdown-multiselect] Error when trying to load module using systemjs (#10)
   
Hi cck0001,Try something like this:var map = {
    …
    'ng2-dropdown-multiselect': 'npm:ng2-dropdown-multiselect'
};

var packages = {
    …
    'ng2-dropdown-multiselect': { main: 'lib/index.js', defaultExtension: 'js', map: {
            './lib/src/components': './lib/src/components/index.js',
            './lib/src/models': './lib/src/models/index.js'
     }}
};This is a working example from my ng2-color-picker app, it may need tweaking slightly. I'll improve integration with systemjs in the future.Regards,Andy—

**Comment by AndyMeps** [comment]
Great to hear you got it working, I'll add these details to the README.md and hopefully i'll be able to simplify implementation for SystemJS in the future.

Thanks again,

Andy

**Comment by Yoten** [comment]
I'm getting the following error when trying to use SystemJS:

> Error: Unable to dynamically transpile ES module. A loader plugin needs to be configured via SystemJS.config({ transpiler: 'transpiler-module' })

This is with a barebones starter project and the systemjs config changes given by cck0001, with v1.2.0's lib-->dist change taken into account.

More specific details, including a link to the starter project I'm using, can be found at this reddit post I made:

https://www.reddit.com/r/Angular2/comments/67yzgo/simple_transpiling_of_es_modules/

I'm new to Angular/Node/Typescript so I originally figured it was a more general mistake I was making rather than something specific to your plugin, but I suppose it doesn't hurt to check.

(My post in #9 is a separate issue, since I gave the CLI a shot after I was first blocked by this one)

**Comment by AndyMeps** [comment]
Hey Yoten,

I'm very conscious of needing to test this on SystemJS as I've been exclusively developing it in ng cli; so I'm really sorry. I'll look in to this later today and try to get this resolved.

Cheers,

Andy

**Comment by cytronn** [comment]
Hi Andy, have you any update on the Systemjs adaptation of the module?
